src/server/ns_turn_maps.c

```diff
@@ -655,16 +668,31 @@ static addr_elem *addr_list_get(addr_list_header *slh, const ioa_addr *key) {
   return NULL;
 }
 
-static const addr_elem *addr_list_get_const(const addr_list_header *slh, const ioa_addr *key) {
+static addr_elem *addr_list_get_no_port(addr_list_header *slh, const ioa_addr *key) {
+  return _addr_list_get(slh, key, 0);
+}
+
+static addr_elem *addr_list_get(addr_list_header *slh, const ioa_addr *key) {
+  return _addr_list_get(slh, key, 1);
+}
+
+static const addr_elem *_addr_list_get_const(const addr_list_header *slh, const ioa_addr *key, int port) {
+
   if (!slh || !key) {
     return NULL;
   }
 
   for (size_t i = 0; i < ADDR_ARRAY_SIZE; ++i) {
     const addr_elem *elem = &(slh->main_list[i]);
     if (elem->value) {
-      if (addr_eq(&(elem->key), key)) {
-        return elem;
+      if (port) {
+        if (addr_eq(&(elem->key), key)) {
+          return elem;
+        }
+      } else {
+        if (addr_eq_no_port(&(elem->key), key)) {
+          return elem;
+        }
       }
```

Comment:
Maybe if we set `addr_eq`/`addr_eq_no_port` to a function pointer and called that instead? Let me know your thoughts.

Reply:
I reworked this so that we no longer need the `*_no_port` variants. Instead we just copy address and use `addr_set_port()` to set the port to 0 before checking the map or storing the address in the map.

Functionally this is the same, so it works great and avoids adding the new code.